caddy.go

```diff
@@ -670,6 +671,85 @@ func Validate(cfg *Config) error {
 	return err
 }
 
+// ExportStorage calls List() and Load() on the configured
+// storage module and returns the underlying assets. 
+func ExportStorage(cfg *Config) (files []struct{
+	Name string
+	Body []byte
+}, err error) {
+	ctx, err := run(cfg, false)
```

Comment:
Rather than running the config, maybe better to load and unmarshal just the `storage` portion. You could try doing something like this:

https://github.com/caddyserver/caddy/blob/cfc85ae8caf032c3ef92675a024fc5d1264ab7ce/cmd/commandfuncs.go#L693-L717

You might need to make a caddy.Context and then call LoadModule() on it though -- I haven't tried doing that in isolation like this but I'd be happy to help if that is problematic...

Reply:
Yeah I did try it this way to easily get at a caddy.Context, I'll give that a shot.